angularSTEP/src/app/upload-recipe/upload-recipe.component.ts

```diff
@@ -92,17 +92,28 @@ export class UploadRecipeComponent {
   }
 
   onSubmit() {
-    this.db.collection('recipes').add({
-      recipeName: this.autoCapitalizeName(this.recipeForm.value.name),
-      description: this.autoCapitalizeFirst(this.recipeForm.value.description),
-      ingredients: this.ingredients,
-      tools: this.tools,
-      instructions: this.instructions,
-      extraInfo: this.autoCapitalizeFirst(this.recipeForm.value.extraInfo),
-      difficulty: this.recipeForm.value.difficulty,
-      timestamp: Date.now(),
-    });
-    
+    if (this.autoCapitalizeFirst(this.recipeForm.value.extraInfo)) {
```

Comment:
Other than the extraInfo, most of these two branches looks nearly identical. Could it be combined more by creating an object, then adding extraInfo into it if needed, then adding to recipes collection?

Reply:
Also this is another good candidate for complex component behavior that could be tested